test/transaction_tracker.go

```diff
@@ -0,0 +1,167 @@
+package test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	ethereum "github.com/celo-org/celo-blockchain"
+	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/core/types"
+	"github.com/celo-org/celo-blockchain/ethclient"
+	"github.com/celo-org/celo-blockchain/event"
+)
+
+var (
+	errStopped = errors.New("transaction tracker closed")
+)
+
+// TransactionTracker tracks processed transactions through a subscription with
+// an ethclient. It provides the ability to check whether transactions have
+// been processed and to wait till those transactions have been processed.
+type TransactionTracker struct {
+	client *ethclient.Client
+	heads  chan *types.Header
+	sub    ethereum.Subscription
+	wg     sync.WaitGroup
+	// processed maps transaction hashes to the block they were processed in.
+	processed   map[common.Hash]*types.Block
+	processedMu *sync.Mutex
+	stopCh      chan struct{}
+	newTxs      event.Feed
+}
+
+// NewTransactionTracker creates a new transaction tracker, it subscribes to
+// new head events on the client and starts processing the events in a
+// goroutine.
+func NewTransactionTracker() *TransactionTracker {
+	return &TransactionTracker{
+		heads:       make(chan *types.Header),
```

Comment:
Subscriptions will block if the channels are full, so we typically use buffered channels.
```suggestion
		heads:       make(chan *types.Header, 10),
```